streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableGlobalKTableJoinTest.java

```diff
@@ -0,0 +1,156 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.kstream.internals;
+
+import org.apache.kafka.streams.processor.Processor;
+import org.apache.kafka.test.KTableValueGetterStub;
+import org.apache.kafka.test.MockKeyValueMapper;
+import org.apache.kafka.test.MockValueJoiner;
+import org.apache.kafka.test.NoOpProcessorContext;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+@SuppressWarnings("unchecked")
+public class KTableGlobalKTableJoinTest {
+
+    private final KTableValueGetterStub<String, String> global = new KTableValueGetterStub<>();
+
+    private final KTableGlobalKTableJoin<String, String, String, String, String> join
+            = new KTableGlobalKTableJoin<>(new ValueGetterSupplier<>(new KTableValueGetterStub<String, String>()),
+                                           new ValueGetterSupplier<>(global),
+                                           MockValueJoiner.STRING_JOINER,
+                                           MockKeyValueMapper.<String, String>SelectValueMapper());
+
+    private NoOpProcessorContext context;
+
+    @Before
+    public void before() {
+        context = new NoOpProcessorContext();
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void shouldNotForwardIfKeyNotInOtherStore() throws Exception {
```

Comment:
I don't agree with the name you have suggested, but i also think the name of the test is not completely correct